lib/optimizely/event_builder.rb

```diff
@@ -195,43 +195,38 @@ def get_conversion_params(event_key, event_tags, experiment_variation_map)
       # event_tags -               +Hash+ Values associated with the event.
       # experiment_variation_map - +Hash+ Map of experiment IDs to bucketed variation IDs
       #
-      # Returns +Hash+ Impression event params
-
-      conversion_event_params = []
+      # Returns +Hash+ Conversion event params
 
+      single_snapshot = {}
+      single_snapshot[:decisions] = []
       experiment_variation_map.each do |experiment_id, variation_id|
-        single_snapshot = {
-          decisions: [{
-            campaign_id: @config.experiment_id_map[experiment_id]['layerId'],
-            experiment_id: experiment_id,
-            variation_id: variation_id
-          }],
-          events: []
-        }
-
-        event_object = {
-          entity_id: @config.event_key_map[event_key]['id'],
-          timestamp: create_timestamp,
-          uuid: create_uuid,
-          key: event_key
-        }
-
-        if event_tags
-          revenue_value = Helpers::EventTagUtils.get_revenue_value(event_tags, @logger)
-          event_object[:revenue] = revenue_value if revenue_value
+        next unless variation_id
+        single_snapshot[:decisions].push(
+          campaign_id: @config.experiment_id_map[experiment_id]['layerId'],
+          experiment_id: experiment_id,
+          variation_id: variation_id
+        )
+      end
 
-          numeric_value = Helpers::EventTagUtils.get_numeric_value(event_tags, @logger)
-          event_object[:value] = numeric_value if numeric_value
+      event_object = {
+        entity_id: @config.event_key_map[event_key]['id'],
+        timestamp: create_timestamp,
+        uuid: create_uuid,
+        key: event_key
+      }
 
-          event_object[:tags] = event_tags
-        end
+      if event_tags
+        revenue_value = Helpers::EventTagUtils.get_revenue_value(event_tags, @logger)
+        event_object[:revenue] = revenue_value if revenue_value
 
-        single_snapshot[:events] = [event_object]
+        numeric_value = Helpers::EventTagUtils.get_numeric_value(event_tags, @logger)
+        event_object[:value] = numeric_value if numeric_value
 
-        conversion_event_params.push(single_snapshot)
+        event_object[:tags] = event_tags
```

Comment:
eventtags should have atleast one value then 'tags' should be added. Please add condition. 
https://github.com/optimizely/python-sdk/pull/136/files#diff-f5fd21234bea2bc5c0c2cbbf5931d93cR276